metrics-runtime/src/data/snapshot.rs

```diff
@@ -1,31 +1,38 @@
 use crate::common::ValueSnapshot;
 use metrics_core::{Key, Recorder, Snapshot as MetricsSnapshot};
-use std::borrow::Cow;
 
 /// A point-in-time view of metric data.
 #[derive(Default, Debug)]
 pub struct Snapshot {
-    measurements: Vec<(String, ValueSnapshot)>,
+    measurements: Vec<(Key, ValueSnapshot)>,
 }
 
 impl Snapshot {
-    pub(crate) fn from(from: Vec<(String, ValueSnapshot)>) -> Self {
+    pub(crate) fn from(from: Vec<(Key, ValueSnapshot)>) -> Self {
```

Comment:
I think it would probably better to use `From<Vec<(Key, ValueSnapshot)>>` here but hide the implementation using `#[doc(hidden)]`.

Reply:
Yeah, I'm fine with generalizing the source data, but why the `#[doc(hidden)]`?  Crate visibility already lets us achieve that.

Reply:
Since your from impl is `pub(crate)` and `Sanpshot` is public, I assumed that you waited to hide it from users. And since `From<..>` is public you would need to hide it from the doc. But disregard that if that was not your intend.

Reply:
I ended up changing `from` to `new`, but leaving it `pub(crate)`.  Adding a proper `From` impl would have meant making `ValueSnapshot` public, which I didn't want to do.